File: tests/python/cases/case_178_local_cache_named_retained_08.py
"""
Task case 178: local_cache_named_retained_08 false-positive/false-negative stress pattern.
"""

RETAINED_CACHE = {}
RETAINED_AUDIT = []
RETAINED_HANDLERS = []

def case_178_local_cache_named_retained_08(input_data):
    task_name = 'local_cache_named_retained_08'
    raw = input_data or 'sample'
    payload = {
        'task': task_name,
        'entity': 'stress',
        'stage': 'evaluation',
        'input': raw,
        'checksum': f'{task_name}:{len(raw)}',
    }
    retained_cache_local = {}
    retained_cache_local['tmp'] = payload
    # SAFE: payload is only stored in local container that dies at return.
    return payload['checksum']

File: tests/nodejs/cases/case_178_local_cache_named_retained_08.js
/**
 * Task case 178: local_cache_named_retained_08 false-positive/false-negative stress pattern.
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case178LocalCacheNamedRetained08(input) {
  const taskName = 'local_cache_named_retained_08';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'stress',
    stage: 'evaluation',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  const retainedCacheLocal = new Map();
  retainedCacheLocal.set('tmp', payload);
  // SAFE: local map does not escape.
  return payload.checksum;
}

module.exports = {
  case178LocalCacheNamedRetained08,
};
